R/MsBackendSql.R

```diff
@@ -702,12 +702,18 @@ setMethod(
 #'
 #' @exportMethod filterRt
 setMethod("filterRt", "MsBackendSql", function(object, rt = numeric(),
-                                              msLevel. = integer()) {
+                                               msLevel. = integer()) {
     if (!length(rt) || all(is.infinite(rt)))
         return(object)
     rt <- range(rt)
-    if (.has_local_variable(object, c("msLevel", "rtime")) |
-        (.has_local_variable(object, "msLevel") & length(msLevel.))) {
+    if (.has_local_variable(object, "rtime") |
+        (length(msLevel.) && .has_local_variable(object, "msLevel"))) {
+        if (length(msLevel.)) {
+            if (!.has_local_variable(object, "msLevel"))
+                object$msLevel <- object$msLevel
```

Comment:
I am against comments for obvious things but this assignment looks useless without background knowledge about `MsBackendCached`. If I am right the getter method access the data storage if needed and assigns the value to the `localData` slot via the setter method. Maybe a like "Move information from data storage (SQL) to local data.frame" would be helpful?

Reply:
yes, will add some comments.